formats/json/jsMain/src/kotlinx/serialization/json/internal/DynamicDecoders.kt

```diff
@@ -81,14 +83,23 @@ private open class DynamicInput(
     override fun decodeElementIndex(descriptor: SerialDescriptor): Int {
         while (currentPosition < descriptor.elementsCount) {
             val name = descriptor.getTag(currentPosition++)
-            if (hasName(name) && (!json.configuration.coerceInputValues || !coerceInputValue(descriptor, currentPosition - 1, name)))
-                return currentPosition - 1
+            val index = currentPosition - 1
+            forceNull = false
```

Comment:
why we had to use `JsonElementMarker` in `TreeJsonDecoder` but not here? Both these decoders are random-access

Reply:
Indeed... hurried. Thanks!)